src/services/core.service.ts

```diff
@@ -32,6 +33,7 @@ export class CoreService {
         this.config.sourceBundle = this.resetSourceBundle();
         this.currentLanguage = vipConfig.language;
         this.currentRegion = vipConfig.region;
+        this.httpOptions = vipConfig.httpOptions;
```

Comment:
L30: only the ‘withCredentials’ passed in httpOptions,  timeout will be overridden.

Reply:
Good concern! The default timeout is set to 0 which indicates no timeout. With that default value, any remote end can keep us waiting for the requested resource for an indefinite period.